python/lsst/ts/salkafka/kafka_producer_factory.py

```diff
@@ -160,12 +184,24 @@ async def make_producer(self, avro_schema):
             schema=avro_schema,
             subject=f"{avro_schema['name']}-value",
         )
-        producer = AIOKafkaProducer(
-            loop=asyncio.get_running_loop(),
-            bootstrap_servers=self.config.broker_url,
-            acks=self.config.wait_for_ack,
-            value_serializer=serializer,
-        )
+        if self.config.sasl_plain_username:
```

Comment:
Here again I suggest checking that both or neither are None before constructing the AIOKafkaProducer.

(A simple function that asserts that both or neither are None might reduce repetition.)